src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -394,7 +410,7 @@ def make_dependency_files(parsed_config, config_file_path, to_stdout):
                             # exists. In that case we save the fallback_entry result
                             # and only use it at the end if nothing more
                             # specific is found.
-                            if not specific_matrices_entry["matrix"]:
+                            if not specific_matrices_entry.get("matrix", None):
```

Comment:
When is the matrix key optional? For a specific entry it should be required, even with the new CLI functionality, right?